sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/options/BlobBatchSetBlobAccessTierOptions.java

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.batch.options;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.blob.BlobUrlParts;
+import com.azure.storage.blob.models.AccessTier;
+import com.azure.storage.blob.models.RehydratePriority;
+import com.azure.storage.common.Utility;
+import com.azure.storage.common.implementation.StorageImplUtils;
+
+/**
+ * Extended options that may be passed when batch setting tier for blobs.
+ */
+@Fluent
+public class BlobBatchSetBlobAccessTierOptions {
+
+    private final ClientLogger logger = new ClientLogger(BlobBatchSetBlobAccessTierOptions.class);
+
+    private final BlobUrlParts blobUrlParts;
+    private final AccessTier tier;
+    private RehydratePriority priority;
+    private String leaseId;
+    private String tagsConditions;
+
+    /**
+     * @param blobUrl Url of the blob to set access tier. Blob names must be encoded to UTF-8.
+     * @param tier {@link AccessTier} to set on each blob.
+     */
+    public BlobBatchSetBlobAccessTierOptions(String blobUrl, AccessTier tier) {
+        StorageImplUtils.assertNotNull("blobUrl", blobUrl);
+        StorageImplUtils.assertNotNull("tier", tier);
+        this.blobUrlParts = BlobUrlParts.parse(blobUrl);
+        this.tier = tier;
+    }
+
+    /**
+     * @param containerName Name of the container to set access tier.
+     * @param blobName Name of the blob to set access tier. Blob names must be encoded to UTF-8.
+     * @param tier {@link AccessTier} to set on each blob.
+     */
+    public BlobBatchSetBlobAccessTierOptions(String containerName, String blobName, AccessTier tier) {
+        StorageImplUtils.assertNotNull("containerName", containerName);
+        StorageImplUtils.assertNotNull("blobName", blobName);
+        StorageImplUtils.assertNotNull("tier", tier);
+        this.blobUrlParts = BlobUrlParts.parse("https://account.blob.core.windows.net")
+            .setContainerName(containerName)
+            .setBlobName(blobName);
+        this.tier = tier;
+    }
+
+    /**
+     * @return Url of the blob to set its access tier.
+     */
+    public String getBlobUrl() {
+        return blobUrlParts.toUrl().toString();
+    }
+
+    /**
+     * @return Container of the blob to set its access tier.
+     */
+    public String getBlobContainerName() {
+        return blobUrlParts.getBlobContainerName();
+    }
+
+    /**
+     * @return Name of the blob to set its access tier.
+     */
+    public String getBlobName() {
+        return blobUrlParts.getBlobName();
+    }
+
+    /**
+     * @return Path of the blob to set its access tier.
+     */
+    public String getBlobPath() {
```

Comment:
I think maybe like `getBlobIdentifier` might be more appropriate since it's not just the path?

Reply:
Yeah I like that